### PR TITLE
fix camel-quarkus-jacksonxml:JsonView annotations take no effect in n…

### PR DESCRIPTION
**Pull request by ffang**
…ative mode #2681

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

**Comment by ppalaga** [comment]
Could perhaps the extension auto-register the required classes?

**Comment by jamesnetherton** [comment]
> Could perhaps the extension auto-register the required classes?

+1 I had the same thought. It'd make the UX better.

**Comment by ppalaga** [comment]
You can use Jandex to find instances of `@JsonView` in the user code. Here is some overview https://peter.palaga.org/presentations/210407-quarkus-insights-how-to-write-quarkus-extensions/index.html#/jandex-example 
The pattern is used several times in Camel Quarkus code base, just grep for `CombinedIndexBuildItem`.

**Comment by ffang** [comment]
Hi @ppalaga , @jamesnetherton ,

Just pushed a new commit which uses jandex to read JsonView annotations and add class to reflection list per your guys suggested.

Thanks!
Freeman

File: extensions/jacksonxml/deployment/src/main/java/org/apache/camel/quarkus/component/jacksonxml/deployment/JacksonxmlProcessor.java [hunk 1]
```diff
@@ -28,4 +37,21 @@ FeatureBuildItem feature() {
...
+    @BuildStep
+    ReflectiveClassBuildItem registerJsonView(CombinedIndexBuildItem combinedIndex) {
+
+        IndexView index = combinedIndex.getIndex();
+        DotName JSON_VIEW = DotName.createSimple(JsonView.class.getName());
+        String[] jsonViews = index.getAnnotations(JSON_VIEW).stream().map(ai -> ai.value())
+                .filter(p -> AnnotationValue.Kind.ARRAY.equals(p.kind()))
+                .map((Function<? super AnnotationValue, ? extends String>) ai -> {
+                    Type[] annotationType = ai.asClassArray();
+                    return annotationType[0].name().toString();
```
Comment (ppalaga) [inline extensions/jacksonxml/deployment/src/main/java/org/apache/camel/quarkus/component/jacksonxml/deployment/JacksonxmlProcessor.java:49]: Why are we taking only the first array element into account? IMO all elements need to be registered for reflection.

Comment (ffang) [reply extensions/jacksonxml/deployment/src/main/java/org/apache/camel/quarkus/component/jacksonxml/deployment/JacksonxmlProcessor.java:49]: You are right, I will send another commit.
Thanks!

Comment (ffang) [reply extensions/jacksonxml/deployment/src/main/java/org/apache/camel/quarkus/component/jacksonxml/deployment/JacksonxmlProcessor.java:49]: Hi @ppalaga ,

And here is the follow up PR
https://github.com/apache/camel-quarkus/pull/2709

Please take a look.
Freeman

**Pull request merged**